### PR TITLE
fix(android): spinner background issues

### LINKED ISSUE #112: Dropdown icon disappears on Android when backgroundColor set
# Bug report

## Summary
Setting `backgroundColor` for `Picker` makes the dropdown icon disappear in android. 

## Environment info

`react-native info` output:

```bash
System:
    OS: Linux 4.15 Ubuntu 18.04.4 LTS (Bionic Beaver)
    CPU: (12) x64 AMD Ryzen 5 1600 Six-Core Processor
    Memory: 8.66 GB / 15.68 GB
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 14.3.0 - /usr/local/bin/node
    Yarn: Not Found
    npm: 6.14.4 - /usr/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    Android SDK:
      API Levels: 23, 28, 30
      Build Tools: 28.0.3, 30.0.0
      System Images: android-28 | Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6514223
  Languages:
    Java: 1.8.0_191 - /usr/bin/javac
    Python: 2.7.17 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0 
    react-native: 0.62.2 => 0.62.2 
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: 1.6.5

## Steps to reproduce

```
<Picker style={{height: 50, width: '50%', alignSelf: 'center', color: '#FAFFFD' backgroundColor: '#393B41'}} />
```
Describe what you expected to happen:

The dropdown icon should not disappear.

### LINKED ISSUE #235: Can't set dropdownIconColor to transparent after new release
### Issue :
I updated the module with the latest version [1.13.1](https://github.com/react-native-picker/picker/releases/tag/v1.13.1) and found a regression in the operation of `dropdownIconColor`. The release was supposed to fix `dropdownIconColor` causing a crash when using string variables. It did but it appears that it has created an another issue. You can no longer set the value of this prop to transparent using hex or rgba.

### LINKED ISSUE #266: Support ColorValue type from RN 0.63
With React Native 0.63, the platform is now able to support [PlatfromColor](https://reactnative.dev/blog/2020/07/06/version-0.63#native-colors-platformcolor-dynamiccolorios) to be used as colors. It would be great to update these colors to use the new `ColorValue` type to match the new typing of colors, so that people who use pickers can take advantage of PlatformColors:

https://github.com/react-native-picker/picker/blob/c6addc279b9ebc27bedd61aa6dc3dbb7cb709869/typings/Picker.d.ts#L9

https://github.com/react-native-picker/picker/blob/c6addc279b9ebc27bedd61aa6dc3dbb7cb709869/typings/Picker.d.ts#L77

### PR DESCRIPTION
**Pull request by mateusz1913**
closes #266
closes #235
closes #112

In order to fix issues with buggy color/style changes to android spinner's background
I created custom background resource which should look identical to default one

It allowed to finally have proper color/background change possibilities.

With default background, if one had applied custom backgroundColor in JS,
it overriden whole background for spinner, which resulted in removing
dropdown icon completely.

With new custom background resource, we can apply color changes directly to
icon or spinner background.

As a side-effect, I introduced possibility to change icon ripple color also,
it is exposed as dropdownIconRippleColor prop

**NOTE**: minSdkVersion was increased to 21 (now it matches with react-native core)

### CODE REVIEW

**Comment by zwh1666258377** [comment]
fix it？please go on

**Comment by zwh1666258377** [comment]
It was a great PR~!

**Review by Naturalclar — APPROVED** [review]
@mateusz1913 LGTM! thank you so much!

**Comment by rurikoaraki** [comment]
Thanks for the fix/support of ColorValue! I just tried integrating this in my repo and found that in this change, while the docs claim that dropdownIconColor is type ColorValue, the type file of defines its type as the return type of processColor, which is different from ColorValue. In order to get this to work, I need to call processColor on the color I pass into dropdownIconColor.

This isn't a huge issue for me, but wanted to give you a heads up on the mismatch.

**Comment by Saadnajmi** [comment]
> Thanks for the fix/support of ColorValue! I just tried integrating this in my repo and found that in this change, while the docs claim that dropdownIconColor is type ColorValue, the type file of defines its type as the return type of processColor, which is different from ColorValue. In order to get this to work, I need to call processColor on the color I pass into dropdownIconColor.
> 
> This isn't a huge issue for me, but wanted to give you a heads up on the mismatch.

I would actually call that an issue, because it implies you should call processColor as a client, and you should not be (that is an internal API) and returns a different object than a `ColorValue`

File: android/src/main/res/drawable/ic_dropdown.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="48dp"
```
Comment (winkelsdorf) [inline android/src/main/res/drawable/ic_dropdown.xml:2]: Not an Android dev, but I think this breaks on API < 23.

See new Bugs opened:
https://github.com/react-native-picker/picker/issues/449
https://github.com/react-native-picker/picker/issues/433
https://github.com/react-native-picker/picker/issues/373
https://github.com/react-native-picker/picker/issues/315

Especially this comment:
https://github.com/react-native-picker/picker/issues/373#issuecomment-1035390969

**Pull request merged**